src/pages/workspaces/workspace/DeleteWorkspaceModal.js

```diff
@@ -26,12 +75,17 @@ const DeleteWorkspaceModal = ({ workspace: { workspace: { namespace, name, bucke
     }
   }
 
+  const isDeleteDisabledFromApps = () => hasApps() && !_.isEmpty(getNonDeletableApps())
```

Comment:
Since this is called repeatedly every render, you should make it a computed value rather than a function and change the places that reference it.
```suggestion
  const isDeleteDisabledFromApps = hasApps() && !_.isEmpty(getNonDeletableApps())
```